src/transformState.js

```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const prop in actions[i].extraData) {
+        state[prop] = actions[i].extraData[prop];
+      }
```

Comment:
Objects already have a method to `assign` other objects to it

Reply:
Yep, I have rewrote it as you said, thanks. Btw, still not sure about my formatting, sorry for that (